tesseract-core/src/lib.rs

```diff
@@ -655,12 +655,24 @@ impl Schema {
         let sort = if let Some(ref s) = query.sort {
             // sort column needs to be named by alias
             // Checking if we are going to sort by RCA or not
-            let measure_name = &s.measure.to_string();
-            if measure_name.contains("RCA") {
-                Some(SortSql {
-                    direction: s.direction.clone(),
-                    column: format!("rca"),
-                })
+            let measure_name = s.measure.to_string();
+            if let Some(ref rca) = query.rca {
+                if measure_name == format!("{} RCA", rca.mea.clone().to_string()){
+                    Some(SortSql {
+                        direction: s.direction.clone(),
+                        column: format!("rca"),
+                    })
+                } else {
```

Comment:
would we be able to potentially avoid duplicating this `else` block by converting the nested `if`s into a single compound if statement e.g. `if X && Y ` instead of `if X { if Y { ...` ... ?

Reply:
We cannot because I need to get the column name of mea inside the RCA

Reply:
In that case, could we maybe rewrite the `if` into a match?      like      ```match query.rca {
                Some(ref rca) => {

                },
                None =>
            }``` could that capture the flow while avoiding the duplication?

Reply:
But we still need to duplicate it in a scenario when we have rca column and we perform a sort on something else that is not in the RCA column it won't work. The fix could be we Just first check if the sort.measure contains the  contains RCA in it or not and if contains we do a strict check

Reply:
I guess I'm really averse to extra nesting and duplication so I'm trying to find a way around this :) thoughts about an approach like this?

```rust
            let rca_mea_name = if query.rca.is_some() {
                 query.rca.as_ref().unwrap().mea.to_string() 
            } else {
                "".to_string()
            };
            if measure_name == format!("{} RCA", rca_mea_name) {
                    Some(SortSql {
                        direction: s.direction.clone(),
                        column: "rca".to_string(),
                    })
            }
```

Does this work? What are your thoughts?

Reply:
I like this idea because it has less complexity of nesting. This should work